### PR TITLE
standardize exports

### PR DESCRIPTION
**Pull request by Goldziher**
# PR Checklist

- [x] Have you followed the guidelines in `CONTRIBUTING.rst`?
- [x] Have you got 100% test coverage on new code?
- [x] Have you updated the prose documentation?
- [x] Have you updated the reference documentation?

### CODE REVIEW

File: starlite/config/logging.py [hunk 1]
```diff
@@ -12,6 +12,18 @@
...
 
+__all__ = (
+    "BaseLoggingConfig",
+    "LoggingConfig",
+    "StructLoggingConfig",
+    "default_logger_factory",
+    "default_structlog_processors",
+    "default_wrapper_class",
+    "get_default_handlers",
+    "get_logger_placeholder",
```
Comment (Goldziher) [inline starlite/config/logging.py:23]:
```suggestion
```

File: starlite/config/logging.py [hunk 2]
```diff
@@ -12,6 +12,18 @@
...
 from starlite.serialization import encode_json
 
+__all__ = (
+    "BaseLoggingConfig",
+    "LoggingConfig",
+    "StructLoggingConfig",
+    "default_logger_factory",
+    "default_structlog_processors",
+    "default_wrapper_class",
+    "get_default_handlers",
```
Comment (Goldziher) [inline starlite/config/logging.py:22]:
```suggestion
```

File: starlite/config/logging.py [hunk 3]
```diff
@@ -12,6 +12,18 @@
 )
 from starlite.serialization import encode_json
 
+__all__ = (
+    "BaseLoggingConfig",
+    "LoggingConfig",
+    "StructLoggingConfig",
+    "default_logger_factory",
+    "default_structlog_processors",
+    "default_wrapper_class",
```
Comment (Goldziher) [inline starlite/config/logging.py:21]:
```suggestion
```

Comment (Goldziher) [inline starlite/config/logging.py:21]:
```suggestion
```

File: starlite/config/logging.py [hunk 4]
```diff
@@ -12,6 +12,18 @@
 )
 from starlite.serialization import encode_json
 
+__all__ = (
+    "BaseLoggingConfig",
+    "LoggingConfig",
+    "StructLoggingConfig",
+    "default_logger_factory",
+    "default_structlog_processors",
```
Comment (Goldziher) [inline starlite/config/logging.py:20]:
```suggestion
```

File: starlite/config/logging.py [hunk 5]
```diff
@@ -12,6 +12,18 @@
 )
 from starlite.serialization import encode_json
 
+__all__ = (
+    "BaseLoggingConfig",
+    "LoggingConfig",
+    "StructLoggingConfig",
+    "default_logger_factory",
```
Comment (Goldziher) [inline starlite/config/logging.py:19]:
```suggestion
```

File: starlite/contrib/htmx/utils.py [hunk 6]
```diff
@@ -1,145 +1,159 @@
...
+    "HTMXHeaders",
+    "get_headers",
+    "get_location_headers",
+    "get_push_url_header",
+    "get_redirect_header",
+    "get_refresh_header",
+    "get_replace_url_header",
+    "get_reswap_header",
+    "get_retarget_header",
+    "get_trigger_event_headers",
```
Comment (Goldziher) [inline starlite/contrib/htmx/utils.py:20]:
```suggestion
```

File: starlite/contrib/htmx/utils.py [hunk 7]
```diff
@@ -1,145 +1,159 @@
...
+__all__ = (
+    "HTMXHeaders",
+    "get_headers",
+    "get_location_headers",
+    "get_push_url_header",
+    "get_redirect_header",
+    "get_refresh_header",
+    "get_replace_url_header",
+    "get_reswap_header",
+    "get_retarget_header",
```
Comment (Goldziher) [inline starlite/contrib/htmx/utils.py:19]:
```suggestion
```

File: starlite/contrib/htmx/utils.py [hunk 8]
```diff
@@ -1,145 +1,159 @@
...
+
+__all__ = (
+    "HTMXHeaders",
+    "get_headers",
+    "get_location_headers",
+    "get_push_url_header",
+    "get_redirect_header",
+    "get_refresh_header",
+    "get_replace_url_header",
+    "get_reswap_header",
```
Comment (Goldziher) [inline starlite/contrib/htmx/utils.py:18]:
```suggestion
```

File: starlite/contrib/htmx/utils.py [hunk 9]
```diff
@@ -1,145 +1,159 @@
...
+from starlite.serialization import encode_json
+
+__all__ = (
+    "HTMXHeaders",
+    "get_headers",
+    "get_location_headers",
+    "get_push_url_header",
+    "get_redirect_header",
+    "get_refresh_header",
+    "get_replace_url_header",
```
Comment (Goldziher) [inline starlite/contrib/htmx/utils.py:17]:
```suggestion
```

File: starlite/contrib/htmx/utils.py [hunk 10]
```diff
@@ -1,145 +1,159 @@
...
+from starlite.exceptions import ImproperlyConfiguredException
+from starlite.serialization import encode_json
+
+__all__ = (
+    "HTMXHeaders",
+    "get_headers",
+    "get_location_headers",
+    "get_push_url_header",
+    "get_redirect_header",
+    "get_refresh_header",
```
Comment (Goldziher) [inline starlite/contrib/htmx/utils.py:16]:
```suggestion
```

File: starlite/contrib/htmx/utils.py [hunk 11]
```diff
@@ -1,145 +1,159 @@
...
+
+from starlite.exceptions import ImproperlyConfiguredException
+from starlite.serialization import encode_json
+
+__all__ = (
+    "HTMXHeaders",
+    "get_headers",
+    "get_location_headers",
+    "get_push_url_header",
+    "get_redirect_header",
```
Comment (Goldziher) [inline starlite/contrib/htmx/utils.py:15]:
```suggestion
```

File: starlite/contrib/htmx/utils.py [hunk 12]
```diff
@@ -1,145 +1,159 @@
...
+from urllib.parse import quote
+
+from starlite.exceptions import ImproperlyConfiguredException
+from starlite.serialization import encode_json
+
+__all__ = (
+    "HTMXHeaders",
+    "get_headers",
+    "get_location_headers",
+    "get_push_url_header",
```
Comment (Goldziher) [inline starlite/contrib/htmx/utils.py:14]:
```suggestion
```

File: starlite/contrib/htmx/utils.py [hunk 13]
```diff
@@ -1,145 +1,159 @@
...
+from typing import TYPE_CHECKING, Any, Callable
+from urllib.parse import quote
+
+from starlite.exceptions import ImproperlyConfiguredException
+from starlite.serialization import encode_json
+
+__all__ = (
+    "HTMXHeaders",
+    "get_headers",
+    "get_location_headers",
```
Comment (Goldziher) [inline starlite/contrib/htmx/utils.py:13]:
```suggestion
```

File: starlite/contrib/htmx/utils.py [hunk 14]
```diff
@@ -1,145 +1,159 @@
...
+from enum import Enum
+from typing import TYPE_CHECKING, Any, Callable
+from urllib.parse import quote
+
+from starlite.exceptions import ImproperlyConfiguredException
+from starlite.serialization import encode_json
+
+__all__ = (
+    "HTMXHeaders",
+    "get_headers",
```
Comment (Goldziher) [inline starlite/contrib/htmx/utils.py:12]:
```suggestion
```

File: starlite/contrib/sqlalchemy_1/config.py [hunk 15]
```diff
@@ -12,6 +12,15 @@
 from starlite.serialization import decode_json, encode_json
 from starlite.utils import AsyncCallable
 
+__all__ = (
+    "SQLAlchemyConfig",
+    "SQLAlchemyEngineConfig",
+    "SQLAlchemySessionConfig",
+    "default_before_send_handler",
```
Comment (Goldziher) [inline starlite/contrib/sqlalchemy_1/config.py:19]:
```suggestion
```

File: starlite/contrib/sqlalchemy_1/config.py [hunk 16]
```diff
@@ -12,6 +12,15 @@
...
+    "serializer",
```
Comment (Goldziher) [inline starlite/contrib/sqlalchemy_1/config.py:20]:
```suggestion
```

File: starlite/datastructures/url.py [hunk 17]
```diff
@@ -8,6 +8,13 @@
 from starlite.parsers import parse_query_string
 from starlite.types import Empty
 
+__all__ = (
+    "Address",
+    "URL",
+    "make_absolute_url",
```
Comment (Goldziher) [inline starlite/datastructures/url.py:14]:
```suggestion
```

File: starlite/dto.py [hunk 18]
```diff
@@ -31,6 +31,9 @@
     is_typed_dict,
 )
 
+__all__ = ("DTO", "DTOFactory", "get_field_type")
```
Comment (Goldziher) [inline starlite/dto.py:34]:
```suggestion
__all__ = ("DTO", "DTOFactory")
```

File: starlite/handlers/utils.py [hunk 19]
```diff
@@ -5,6 +5,9 @@
 from starlite.datastructures.cookie import Cookie
 from starlite.datastructures.response_header import ResponseHeader
 
+__all__ = ("narrow_response_cookies", "narrow_response_headers")
```
Comment (Goldziher) [inline starlite/handlers/utils.py:8]: these utils are used on different layers of the app - should they be here or in an "internal_utils.py" package? @starlite-api/maintainers

File: starlite/middleware/csrf.py [hunk 20]
```diff
@@ -16,6 +16,9 @@
     should_bypass_middleware,
 )
 
+__all__ = ("CSRFMiddleware", "generate_csrf_hash", "generate_csrf_token")
```
Comment (Goldziher) [inline starlite/middleware/csrf.py:19]:
```suggestion
__all__ = ("CSRFMiddleware")
```

File: starlite/contrib/opentelemetry/__init__.py [hunk 21]
```diff
@@ -1,5 +1,5 @@
+from ._utils import get_route_details_from_scope
 from .config import OpenTelemetryConfig
 from .middleware import OpenTelemetryInstrumentationMiddleware
-from .utils import get_route_details_from_scope
 
 __all__ = ("OpenTelemetryConfig", "OpenTelemetryInstrumentationMiddleware", "get_route_details_from_scope")
```
Comment (Goldziher) [inline starlite/contrib/opentelemetry/__init__.py:5]:
```suggestion
__all__ = ("OpenTelemetryConfig", "OpenTelemetryInstrumentationMiddleware")
```

File: starlite/types/file_types.py [hunk 22]
```diff
@@ -13,6 +13,9 @@
 
 from typing_extensions import NotRequired, TypedDict
 
+__all__ = ["FileInfo", "FileSystemProtocol"]
```
Comment (peterschutt) [inline starlite/types/file_types.py:16]: should be tuple

File: starlite/types/internal_types.py [hunk 23]
```diff
@@ -12,6 +12,9 @@
 
 from starlite.types import Method
 
+__all__ = ["PathParameterDefinition"]
```
Comment (peterschutt) [inline starlite/types/internal_types.py:15]: should be tuple

File: starlite/utils/dataclass.py [hunk 24]
```diff
@@ -2,6 +2,9 @@
 
 from typing import TYPE_CHECKING, Any, Iterable, cast
 
+__all__ = ["extract_dataclass_fields"]
```
Comment (peterschutt) [inline starlite/utils/dataclass.py:5]: should be tuple

File: starlite/utils/deprecation.py [hunk 25]
```diff
@@ -7,6 +7,9 @@
 
 from typing_extensions import ParamSpec
 
+__all__ = ["deprecated", "warn_deprecation"]
```
Comment (peterschutt) [inline starlite/utils/deprecation.py:10]: shoudl be tuple

File: starlite/utils/helpers.py [hunk 26]
```diff
@@ -4,6 +4,9 @@
 from enum import Enum
 from typing import TYPE_CHECKING, Any, Generic, TypeVar, cast
 
+__all__ = ["Ref", "get_enum_string_value", "get_name", "unwrap_partial"]
```
Comment (peterschutt) [inline starlite/utils/helpers.py:7]: should be tuple

File: starlite/utils/path.py [hunk 27]
```diff
@@ -3,6 +3,9 @@
 import re
 from typing import Iterable
 
+__all__ = ["join_paths", "normalize_path"]
```
Comment (peterschutt) [inline starlite/utils/path.py:6]: tuple

File: starlite/utils/pydantic.py [hunk 28]
```diff
@@ -5,6 +5,9 @@
 from pydantic import BaseConfig, BaseModel, create_model, create_model_from_typeddict
 from pydantic_factories.utils import create_model_from_dataclass
 
+__all__ = ["Config", "convert_dataclass_to_model", "convert_typeddict_to_model", "create_parsed_model_field"]
```
Comment (peterschutt) [inline starlite/utils/pydantic.py:8]: tuple

File: starlite/utils/scope.py [hunk 29]
```diff
@@ -4,6 +4,9 @@
 
 from starlite.constants import SCOPE_STATE_NAMESPACE
 
+__all__ = ["get_serializer_from_scope", "get_starlite_scope_state", "set_starlite_scope_state"]
```
Comment (peterschutt) [inline starlite/utils/scope.py:7]: tuple

File: starlite/utils/sequence.py [hunk 30]
```diff
@@ -2,6 +2,9 @@
 
 from typing import Callable, Iterable, TypeVar
 
+__all__ = ["find_index", "unique"]
```
Comment (peterschutt) [inline starlite/utils/sequence.py:5]: tuple

File: starlite/utils/sync.py [hunk 31]
```diff
@@ -19,6 +19,9 @@
 
 from starlite.utils.helpers import Ref
 
+__all__ = ["AsyncCallable", "AsyncIteratorWrapper", "as_async_callable_list", "async_partial", "is_async_callable"]
```
Comment (peterschutt) [inline starlite/utils/sync.py:22]: tuple

File: starlite/utils/types.py [hunk 32]
```diff
@@ -20,6 +20,9 @@
 from starlite.types.builtin_types import NoneType
 from starlite.utils.predicates import is_class_and_subclass
 
+__all__ = ["annotation_is_iterable_of_type", "make_non_optional_union"]
```
Comment (peterschutt) [inline starlite/utils/types.py:23]: tuple

File: starlite/utils/compat.py [hunk 33]
```diff
@@ -27,20 +30,20 @@ async def async_next(gen: AsyncGenerator[T, Any], default: D | EmptyType = Empty
 
 
 @contextmanager
-def py_38_safe_annotations(annotated: Any) -> Generator[Any, None, None]:
-    """Ensure annotations are backward compatible with Python 3.8.
+def py_39_safe_annotations(annotated: Any) -> Generator[Any, None, None]:
```
Comment (peterschutt) [inline starlite/utils/compat.py:33]: good catch!

**Review by peterschutt — CHANGES_REQUESTED** [review]
For consistency, `CacheConfig`, `StaticFilesConfig`, and `TemplateConfig` should be moved into their owning namespace package/module, like `LoggingConfig`.

File: starlite/handlers/http_handlers/_utils.py [hunk 34]
```diff
@@ -2,7 +2,7 @@
 
 from functools import lru_cache
 from inspect import isawaitable
-from typing import TYPE_CHECKING, cast
+from typing import TYPE_CHECKING, Any, Sequence, cast
```
[the remark this answers is not in the document]
Comment (Goldziher) [reply starlite/handlers/http_handlers/_utils.py:5]: its what we have everywhere, we need to reconfigure the plugin to reject these first

**Review by peterschutt — APPROVED** [review]

**Pull request merged**